### PR TITLE
Test: Upgrade test, delete kube-dns pods before install.

### PR DESCRIPTION
**Pull request by eloycoto**
Delete Kube-dns pods before cilium 1.0 is installed. This is because if
not cilium 1.0 try to restore pods from 1.1 and maybe it fails.

Related #4329

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4675)

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by eloycoto** [comment]
@aanm it does not matter if the kube-dns is deleted before the install: 

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/432/testReport/junit/k8s-1/8/K8sValidatedUpdates_Updating_Cilium_stable_to_master/
[f0c5341e_K8sValidatedUpdates_Updating_Cilium_stable_to_master(6).zip](https://github.com/cilium/cilium/files/2148010/f0c5341e_K8sValidatedUpdates_Updating_Cilium_stable_to_master.6.zip)

```
Stacktrace

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
DNS entry is not ready after timeout
Expected
    <*errors.errorString | 0xc4202aa3a0>: {
        s: "Timeout reached: DNS \"app1-service.default.svc.cluster.local\" is not ready after timeout",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/k8sT/Updates.go:122

Standard Output

STEP: Checking that installed image is "cilium/cilium:v1.0.0"
STEP: Installing kube-dns
STEP: Creating some endpoints and L7 policy
=== Test Finished at 2018-06-28T14:29:44Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
default       app1-7796cc66f4-qm72s              1/1       Running   0          8m        10.10.0.32      k8s1
default       app1-7796cc66f4-rdkvz              1/1       Running   0          8m        10.10.0.59      k8s1
default       app2-7dbb458fcc-qgrgn              1/1       Running   0          8m        10.10.0.66      k8s1
default       app3-787bfc6d9f-26qv4              1/1       Running   0          8m        10.10.0.142     k8s1
kube-system   cilium-8kd5l                       1/1       Running   0          9m        192.168.36.12   k8s2
kube-system   cilium-fcsxh                       1/1       Running   0          9m        192.168.36.11   k8s1
kube-system   etcd-k8s1                          1/1       Running   0          44m       192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1                1/1       Running   0          43m       192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1       1/1       Running   0          43m       192.168.36.11   k8s1
kube-system   kube-dns-6f4c8dbcd4-2fw62          3/3       Running   0          8m        10.10.1.66      k8s2
kube-system   kube-proxy-44b72                   1/1       Running   0          37m       192.168.36.12   k8s2
kube-system   kube-proxy-fqxsx                   1/1       Running   0          44m       192.168.36.11   k8s1
kube-system   kube-scheduler-k8s1                1/1       Running   0          43m       192.168.36.11   k8s1
prometheus    prometheus-core-5c9fbf5cc6-xdxg2   1/1       Running   0          37m       10.10.0.134     k8s1

cmd: kubectl exec -n kube-system cilium-8kd5l -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                   IPv6                   IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                     
173        Disabled           Disabled          55437      k8s:io.kubernetes.pod.namespace=kube-system   f00d::a0a:100:0:ad     10.10.1.66    ready   
                                                           k8s:k8s-app=kube-dns                                                                       
29898      Disabled           Disabled          56968      reserved:health                               f00d::a0a:100:0:74ca   10.10.1.172   ready   

cmd: kubectl exec -n kube-system cilium-fcsxh -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                  IPv6                 IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                  
173        Disabled           Disabled          63759      k8s:id=app2                                  f00d::a0a:0:0:ad     10.10.0.66    ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                 
                                                           k8s:zgroup=testapp                                                                      
13949      Enabled            Disabled          59960      k8s:id=app1                                  f00d::a0a:0:0:367d   10.10.0.32    ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                 
                                                           k8s:zgroup=testapp                                                                      
27131      Disabled           Disabled          53004      k8s:app=prometheus                           f00d::a0a:0:0:69fb   10.10.0.134   ready   
                                                           k8s:component=core                                                                      
                                                           k8s:io.kubernetes.pod.namespace=prometheus                                              
29381      Enabled            Disabled          59960      k8s:id=app1                                  f00d::a0a:0:0:72c5   10.10.0.59    ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                 
                                                           k8s:zgroup=testapp                                                                      
29898      Disabled           Disabled          56968      reserved:health                              f00d::a0a:0:0:74ca   10.10.0.172   ready   
48896      Disabled           Disabled          64198      k8s:id=app3                                  f00d::a0a:0:0:bf00   10.10.0.142   ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                 
                                                           k8s:zgroup=testapp                                                                      

===================== Existing AfterFailed =====================
```

The DNS status: 

#### BPF Endpoint List 173

```
IP ADDRESS             LOCAL ENDPOINT INFO                                                                                     
10.10.1.66             id=173   ifindex=95  mac=00:00:00:00:9E:FB nodemac=C3:3F:00:00:86:FF seclabel=0xd88d portMaps=(empty)   
10.10.1.44             (localhost)                                                                                             
f00d::a0a:100:0:ad     id=173   ifindex=95  mac=00:00:00:00:9E:FB nodemac=C3:3F:00:00:86:FF seclabel=0xd88d portMaps=(empty)   
f00d::a0a:100:0:8ad6   (localhost)                                                                                             
fd04::12               (localhost)                                                                                             
10.10.1.172            id=29898 ifindex=93  mac=00:00:00:00:BE:51 nodemac=1B:44:00:00:86:5E seclabel=0xde88 portMaps=(empty)   
192.168.36.12          (localhost)                                                                                             
f00d::a0a:100:0:74ca   id=29898 ifindex=93  mac=00:00:00:00:BE:51 nodemac=1B:44:00:00:86:5E seclabel=0xde88 portMaps=(empty)   

```


#### BPF Policy Get 173

```
DIRECTION   LABELS (source:key[=value])   PORT/PROTO   BYTES   PACKETS   
Ingress     reserved:host                 ANY          0       0         

```


#### BPF CT List 173

```
Error: Unable to open /sys/fs/bpf/tc/globals/cilium_ct6_173: Unable to get object /sys/fs/bpf/tc/globals/cilium_ct6_173: no such file or directory

```


#### BPF LB List 173

```
SERVICE ADDRESS     BACKEND ADDRESS          
10.99.170.97:9090   0.0.0.0:0 (0)            
                    10.10.0.134:9090 (1)     
10.96.0.10:53       0.0.0.0:0 (0)            
                    10.10.1.66:53 (3)        
10.98.229.39:80     10.10.0.32:80 (20)       
                    10.10.0.59:80 (20)       
                    0.0.0.0:0 (0)            
10.96.0.1:443       0.0.0.0:0 (0)            
                    192.168.36.11:6443 (2)   

```


#### BPF Tunnel List 173

```
TUNNEL              VALUE           
10.10.0.0           192.168.36.11   
f00d::a0a:100:0:0   192.168.36.12   
10.10.1.0           192.168.36.12   
f00d::a0a:0:0:0     192.168.36.11   

```


#### Endpoint Get 173

```
[
  {
    "id": 173,
    "spec": {
      "label-configuration": {
        "user": []
      },
      "options": {
        "Conntrack": "Enabled",
        "ConntrackAccounting": "Enabled",
        "ConntrackLocal": "Disabled",
        "Debug": "Enabled",
        "DebugLB": "Disabled",
        "DropNotification": "Enabled",
        "EgressPolicy": "Disabled",
        "IngressPolicy": "Disabled",
        "NAT46": "Disabled",
        "TraceNotification": "Enabled"
      }
    },
    "status": {
      "controllers": [
        {
          "configuration": {
            "error-retry": true,
            "interval": "5m0s"
          },
          "name": "resolve-identity-173",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2018-06-28T14:25:56.467Z",
            "success-count": 2
          },
          "uuid": "78e84ecc-7ade-11e8-9dcc-080027db3f15"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "5m0s"
          },
          "name": "sync-IPv4-identity-mapping (173)",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2018-06-28T14:25:56.476Z",
            "success-count": 2
          },
          "uuid": "78e9196c-7ade-11e8-9dcc-080027db3f15"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "5m0s"
          },
          "name": "sync-IPv6-identity-mapping (173)",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2018-06-28T14:25:56.476Z",
            "success-count": 2
          },
          "uuid": "78e919eb-7ade-11e8-9dcc-080027db3f15"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "1m0s"
          },
          "name": "sync-identity-to-k8s-pod (173)",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2018-06-28T14:28:56.605Z",
            "success-count": 9
          },
          "uuid": "78e91899-7ade-11e8-9dcc-080027db3f15"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "10s"
          },
          "name": "sync-to-k8s-ciliumendpoint (173)",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2018-06-28T14:29:46.951Z",
            "success-count": 54
          },
          "uuid": "788524fb-7ade-11e8-9dcc-080027db3f15"
        }
      ],
      "external-identifiers": {
        "container-id": "31d115dc7834b23592adc70b8f04f48077a44bcd801c39e6790846d45a995693",
        "container-name": "k8s_POD_kube-dns-6f4c8dbcd4-2fw62_kube-system_773557d3-7ade-11e8-a016-080027db3f15_0",
        "pod-name": "kube-system:kube-dns-6f4c8dbcd4-2fw62"
      },
      "health": {
        "bpf": "OK",
        "connected": true,
        "overallHealth": "OK",
        "policy": "OK"
      },
      "identity": {
        "id": 55437,
        "labels": [
          "k8s:io.kubernetes.pod.namespace=kube-system",
          "k8s:k8s-app=kube-dns"
        ],
        "labelsSHA256": "1b2bafe403e0d433f8307836b07ade2cf802511501f880917e128b7f663ac313"
      },
      "labels": {
        "derived": [
          "container:annotation.kubernetes.io/config.seen=2018-06-28T14:20:53.896026433Z",
          "container:annotation.kubernetes.io/config.source=api",
          container:annotation.kubernetes.io/created-by={
            "apiVersion": "v1",
            "kind": "SerializedReference",
            "reference": {
              "apiVersion": "extensions",
              "kind": "ReplicaSet",
              "name": "kube-dns-6f4c8dbcd4",
              "namespace": "kube-system",
              "resourceVersion": "5784",
              "uid": "77344bfd-7ade-11e8-a016-080027db3f15"
            }
          }
,
          "container:io.kubernetes.container.name=POD",
          "container:io.kubernetes.docker.type=podsandbox",
          "container:io.kubernetes.pod.name=kube-dns-6f4c8dbcd4-2fw62",
          "container:io.kubernetes.pod.uid=773557d3-7ade-11e8-a016-080027db3f15",
          "k8s:pod-template-hash=2907486780"
        ],
        "disabled": [],
        "realized": {
          "user": []
        },
        "security-relevant": [
          "k8s:io.kubernetes.pod.namespace=kube-system",
          "k8s:k8s-app=kube-dns"
        ]
      },
      "log": [
        {
          "code": "OK",
          "message": "Successfully regenerated endpoint program due to endpoint policy updated & changes were needed",
          "state": "ready",
          "timestamp": "2018-06-28T14:25:43Z"
        }
      ],
      "networking": {
        "addressing": [
          {
            "ipv4": "10.10.1.66",
            "ipv6": "f00d::a0a:100:0:ad"
          }
        ],
        "host-mac": "86:ff:0d:c8:3b:80",
        "interface-index": 95,
        "interface-name": "lxc31d11",
        "mac": "9e:fb:fb:e2:c3:3f"
      },
      "policy": {
        "proxy-statistics": [],
        "realized": {
          "allowed-egress-identities": [],
          "allowed-ingress-identities": [
            1
          ],
          "build": 13,
          "cidr-policy": {
            "egress": [],
            "ingress": []
          },
          "id": 55437,
          "l4": {
            "egress": [],
            "ingress": []
          },
          "policy-enabled": "none",
          "policy-revision": 13
        },
        "spec": {
          "allowed-egress-identities": [],
          "allowed-ingress-identities": [
            1
          ],
          "build": 13,
          "cidr-policy": {
            "egress": [],
            "ingress": []
          },
          "id": 55437,
          "l4": {
            "egress": [],
            "ingress": []
          },
          "policy-enabled": "none",
          "policy-revision": 13
        }
      },
      "realized": {
        "label-configuration": {
          "user": []
        },
        "options": {
          "Conntrack": "Enabled",
          "ConntrackAccounting": "Enabled",
          "ConntrackLocal": "Disabled",
          "Debug": "Enabled",
          "DebugLB": "Disabled",
          "DropNotification": "Enabled",
          "EgressPolicy": "Disabled",
          "IngressPolicy": "Disabled",
          "NAT46": "Disabled",
          "TraceNotification": "Enabled"
        }
      },
      "state": "ready"
    }
  }
]


```


#### Endpoint Log 173

```
Timestamp              Status    State                   Message
2018-06-28T14:25:43Z   OK        ready                   Successfully regenerated endpoint program due to endpoint policy updated & changes were needed
2018-06-28T14:25:43Z   OK        ready                   Completed endpoint regeneration with no pending regeneration requests
2018-06-28T14:25:41Z   OK        regenerating            Regenerating Endpoint BPF: endpoint policy updated & changes were needed
2018-06-28T14:25:41Z   OK        waiting-to-regenerate   Triggering endpoint regeneration due to policy updates
2018-06-28T14:21:13Z   OK        ready                   Successfully regenerated endpoint program due to endpoint policy updated & changes were needed
2018-06-28T14:21:13Z   OK        ready                   Completed endpoint regeneration with no pending regeneration requests
2018-06-28T14:21:10Z   OK        regenerating            Regenerating Endpoint BPF: endpoint policy updated & changes were needed
2018-06-28T14:21:10Z   OK        waiting-to-regenerate   Triggering endpoint regeneration due to policy updates
2018-06-28T14:21:03Z   OK        ready                   Successfully regenerated endpoint program due to endpoint policy updated & changes were needed
2018-06-28T14:21:03Z   OK        ready                   Completed endpoint regeneration with no pending regeneration requests
2018-06-28T14:21:01Z   OK        regenerating            Regenerating Endpoint BPF: endpoint policy updated & changes were needed
2018-06-28T14:21:01Z   OK        waiting-to-regenerate   Successfully regenerated endpoint program due to endpoint policy updated & changes were needed
2018-06-28T14:21:01Z   Warning   waiting-to-regenerate   Skipped invalid state transition to ready due to: Completed endpoint regeneration with no pending regeneration requests
2018-06-28T14:21:00Z   Warning   waiting-to-regenerate   Skipped invalid state transition to waiting-to-regenerate due to: Triggering endpoint regeneration due to policy updates
2018-06-28T14:21:00Z   OK        waiting-to-regenerate   Triggering endpoint regeneration due to policy updates
2018-06-28T14:20:59Z   OK        regenerating            Regenerating Endpoint BPF: endpoint policy updated & changes were needed
2018-06-28T14:20:59Z   OK        waiting-to-regenerate   Triggering endpoint regeneration due to policy updates
2018-06-28T14:20:58Z   OK        ready                   Successfully regenerated endpoint program due to updated security labels
2018-06-28T14:20:58Z   OK        ready                   Completed endpoint regeneration with no pending regeneration requests
2018-06-28T14:20:56Z   OK        regenerating            Regenerating Endpoint BPF: updated security labels
2018-06-28T14:20:56Z   OK        waiting-to-regenerate   Triggering regeneration due to new identity
2018-06-28T14:20:56Z   OK        ready                   Set identity for this endpoint
2018-06-28T14:20:56Z   OK        waiting-for-identity    Endpoint policy update skipped because no changes were needed
2018-06-28T14:20:55Z   OK        waiting-for-identity    Successfully regenerated endpoint program due to Create endpoint from API PUT
2018-06-28T14:20:55Z   Warning   waiting-for-identity    Skipped invalid state transition to ready due to: Completed endpoint regeneration with no pending regeneration requests

```

**Comment by aanm** [comment]
@eloycoto debugging that problem will depend on https://github.com/cilium/cilium/pull/4686

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Review the output, there is communication from APP2 to the DNS: 

```
k8s2: <- overlay flow 0x9e9ae694 identity 0->0 state new ifindex cilium_vxlan: 10.10.0.66:45244 -> 10.10.1.72:53 udp
k8s2: -> endpoint 17318 flow 0x9e9ae694 identity 23964->13306 state new ifindex lxc3571f: 10.10.0.66:45244 -> 10.10.1.72:53 udp
k8s2: <- overlay flow 0x9e9ae694 identity 0->0 state new ifindex cilium_vxlan: 10.10.0.66:45244 -> 10.10.1.72:53 udp
k8s2: -> endpoint 17318 flow 0x9e9ae694 identity 23964->13306 state established ifindex lxc3571f: 10.10.0.66:45244 -> 10.10.1.72:53 udp
k8s2: <- endpoint 17318 flow 0x9e9ae694 identity 13306->0 state new ifindex 0: 10.10.1.72:53 -> 10.10.0.66:45244 udp
k8s2: -> overlay flow 0x9e9ae694 identity 13306->0 state new ifindex cilium_vxlan: 10.10.1.72:53 -> 10.10.0.66:45244 udp
k8s2: <- endpoint 17318 flow 0x9e9ae694 identity 13306->0 state new ifindex 0: 10.10.1.72:53 -> 10.10.0.66:45244 udp
k8s2: -> overlay flow 0x9e9ae694 identity 13306->0 state new ifindex cilium_vxlan: 10.10.1.72:53 -> 10.10.0.66:45244 udp
```

Core-dns does not have any good DNS log, so I'll wait until microscope has timestamps to be able to debug this one. 

[f0c5341e_K8sValidatedUpdates_Updating_Cilium_stable_to_master.zip](https://github.com/cilium/cilium/files/2163741/f0c5341e_K8sValidatedUpdates_Updating_Cilium_stable_to_master.zip)

**Comment by aanm** [comment]
@eloycoto can we enable debug logs in coredns?

Can we merge this as this seems to be hit by a flake? I'm going to trigger a new missed-k8s to make sure it's a flake

**Comment by aanm** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

File: test/provision/manifest/coredns_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
+apiVersion: extensions/v1beta1
```
Comment (aanm) [inline test/provision/manifest/coredns_deployment.yaml:4]: Can you add a header here specifying where we got this file from?

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**